### PR TITLE
docs: Update Segmented Control (Preview) docs

### PR DESCRIPTION
**Pull request by jamesfan**
## Summary

Updates docs for the new Segmented Control in Preview.

## Release Category

Documentation

---

## Checklist

- [x] MDX documentation adheres to Canvas Kit's [standard MDX template](https://github.com/Workday/canvas-kit/discussions/1131)

## For the Reviewer

- [x] PR title is short and descriptive
- [x] PR summary describes the change (Fixes/Resolves linked correctly)
- [x] PR Release Notes describes additional information useful to call out in a release message or removed if not applicable
- [x] Breaking Changes provides useful information to upgrade to this code or removed if not applicable

## Where Should the Reviewer Start?

Review MDX docs for Segmented Control in Preview and corresponding section in v8 Upgrade Guide.

## Areas for Feedback? (optional)

- [x] Documentation

### CODE REVIEW

File: modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx [hunk 1]
```diff
@@ -359,30 +359,23 @@ text with built-in support for our Canvas type tokens.
...
+The `SegmentedControl` in Preview has the following major differences compared to the
+`SegmentedControl` in Main:
+
+- Visually redesigned to better align with the Canvas Design System
+- Uses title casing (e.g., "In Progress") instead of full caps ("IN PROGRESS")
+- Supports a text-only variation
+- Includes built-in support for tooltips to improve accessibility of the icon-only variation
+
+You may still use `SegmentedControl` in the Main package, but note that it will be replaced by
+`SegmentedControl` in the Preview package in the future.
```
Comment (jamesfan) [inline modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx:378]: @RayRedGoose Can you add a code snippet here (between the list and "You may still...") similar to what Manny did below for Status Indicator showing how equivalent Segmented Controls would look in Main vs. Preview?

File: modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx [hunk 2]
```diff
@@ -38,62 +37,86 @@ yarn add @workday/canvas-kit-preview-react
...
-are specified for size (as `medium`) and orientation (as `horizontal`). All buttons should have the
-same type and only one of them can be active at a time.
-
 `SegmentedControl` includes a container `SegmentedControl` component and the following
-subcomponents: `SegmentedControl.List` which contains a `SegmentedControl.Item`.
+subcomponents: `SegmentedControl.List` and `SegmentedControl.Item`.
+
+The example below contains a `SegmentedControl` with four icon-only buttons. Each button is rendered
+using a `SegmentedControl.Item` and is paired with a tooltip describing the button's function. Only
+one button can be active at a time.
```
Comment (jamesfan) [inline modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx:45]: @RayRedGoose Can you update the Basic example to show how a user would extract the currently selected value from the Segmented Control? The example has four buttons for Table, List, Detail, and Diagram -- how do I (as a developer) get the currently selected value? We can just print it out in a paragraph below the example like we do for [Menu](https://5d854c26ba934e0020f5e98a-khqkejkfmy.chromatic.com/?path=/docs/components-popups-menu-react--basic).

File: modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx [hunk 3]
```diff
@@ -38,62 +37,86 @@ yarn add @workday/canvas-kit-preview-react
...
-### Variations Example
+Note that you must provide `SegmentedControl.List` with an `aria-label` prop for accessibility
+reasons.
 
-`SegmentedControl` can be used with diferent buttons: text only, text with icon or icon only.
+### Variations
 
-#### Text And Icon
+`SegmentedControl` supports three variations based on whether or not its `SegmentedControl.Item`
+components have an `icon` prop and/or text content: icon-only, text-only, and text-and-icon.
```
Comment (jamesfan) [inline modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx:55]: I updated the `Variations` section to provide more explanation about how to specify each variation.

File: modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx [hunk 4]
```diff
@@ -147,7 +171,7 @@ underlying `Stack` element.
 
 #### Usage
 
-`SegmentedControl.Item` is a `button` element built on `BaseButton`. `SegmentedControl.Item` has
+`SegmentedControl.Item` is a `button` element built on `BaseButton`. `SegmentedControl.Item` has a
 `data-id` prop to handle `onSelect` properly.
```
Comment (mannycarrera4) [inline modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx:175]: @RayRedGoose did we delete this?

File: modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx [hunk 5]
```diff
@@ -38,62 +37,86 @@ yarn add @workday/canvas-kit-preview-react
...
-example, our items are stored as an array in the state.
+`SegmentedControl` supports a
+[dynamic API](/getting-started/for-developers/resources/collection-api/#dynamic-items) where instead
+of statically providing the JSX for each `SegmentedControl.Item`, you pass an array of `items` in
+the `model` state and provide a render function to display the items.
 
 <ExampleCodeBlock code={Dynamic} />
 
+Note that `SegmentedControl.Item` accepts an optional `index` prop if you want to specify the
+position of the item. Items are appended to the end by default.
```
Comment (jamesfan) [inline modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx:118]: @RayRedGoose I was confused a bit by this `index` prop on `SegmentedControl.Item` (I noticed you lifted this language from the Tabs docs, perhaps we need to revisit those docs as well). I couldn't get it to work with the static API or the dynamic API -- could you provide a code snippet or example illustrating how to use it?

**Review by mannycarrera4 — COMMENTED** [review]
this looks good to me!

**Review by jamesfan — COMMENTED** [review]
@RayRedGoose Nice work on the docs! I wordsmithed the content a bit to add some clarity and to fix some grammar.

Left a few comments below where I could use your help with some improvements.

File: modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx [hunk 6]
```diff
@@ -23,10 +23,9 @@ import {
 
 Segmented Control is a
 [compound component](/getting-started/for-developers/resources/compound-components/) that represents
-a linear group of two, three or four buttons allowing the selection of a specific value. This is a
-controlled component.
+a linear group of multiple buttons allowing the selection of a specific value.
```
Comment (RayRedGoose) [inline modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx:26]:
```suggestion
a linear group of multiple buttons allowing the selection of a specific value. We strongly encourage to use no more than 4 buttons.
```

File: modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx [hunk 7]
```diff
@@ -23,10 +23,9 @@ import {
...
 
-[> Workday Design Reference](https://design.workday.com/components/buttons/action-bar)
+[> Workday Design Reference](https://design.workday.com/components/buttons/segmented-control)
```
Comment (willklein) [inline modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx:28]: Does this get rewritten somewhere or do we need to fix the domain?

File: modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx (hunk 6 above)
Comment (jamesfan) [reply modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx:26]: Got it, I was wondering if the inclusion of that limit was intentional.

File: modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx (hunk 7 above)
Comment (willklein) [reply modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx:28]: Looks like I end up at: https://canvas.workday.com/components/overview/

Comment (NicholasBoll) [reply modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx:28]: https://canvas.workday.com/components/buttons/segmented-control

But depending on if you're internal or external, you'll get redirected

Comment (jamesfan) [reply modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx:28]: It should automatically take you to the correct place (https://canvas.workdaydesign.com/components/buttons/segmented-control) eventually, but there are some redirect issues we need to work out on the site.

File: modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx (hunk 1 above)
Comment (RayRedGoose) [reply modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx]: Added.

File: modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx (hunk 2 above)
Comment (RayRedGoose) [reply modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx:46]: Good point! Added.

File: modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx (hunk 3 above)
Comment (RayRedGoose) [reply modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx:56]: Looks much better! Thank you.

**Review by RayRedGoose — APPROVED** [review]
Great refactoring!

File: modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx (hunk 6 above)
Comment (jamesfan) [reply modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx:26]: FYI: I moved this text to the `Basic Example` section. The paragraph at the top of the page (underneath the H1) is automatically stripped out by the Canvas Site so this guidance won't be visible on the Site unless we move it further down.

File: modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx (hunk 7 above)
Comment (willklein) [reply modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx:28]: Got it!

**Review by NicholasBoll — APPROVED** [review]

**Pull request merged**